### PR TITLE
feat(ci): split jobs

### PR DESCRIPTION
**Pull request by soapwaster**

### CODE REVIEW

**Review by angelocatalani — CHANGES_REQUESTED** [review]
i think it is possible to reuse the steps

**Comment by soapwaster** [comment]
lgty ?

File: .github/actions/poetry/action.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: 'Poetry Install'
+description: 'Poetry install'
```
Comment (angelocatalani) [inline .github/actions/poetry/action.yaml:2]:
```suggestion
description: 'Build the application with Poetry'
```

File: .github/actions/poetry/action.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: 'Poetry Install'
```
Comment (angelocatalani) [inline .github/actions/poetry/action.yaml:1]:
```suggestion
name: 'Build application'
```

Comment (angelocatalani) [reply .github/actions/poetry/action.yaml:1]: I think it is better to rename the subfolder to application and the the yaml to build.yaml

File: .github/actions/poetry/action.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          with:
+            path: .venv
+            key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
+            restore-keys: |
+              ${{ runner.os }}-poetry-
+
+        - name: Install dependecies
+          if: steps.poetry-cache.outputs.cache-hit != 'true'
+          run: poetry install --no-interaction --no-root
+          shell: bash
```
Comment (angelocatalani) [inline .github/actions/poetry/action.yaml:36]: missing new line

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -7,47 +7,30 @@ on:
...
-          restore-keys: |
-            ${{ runner.os }}-poetry-
-
-      - name: Install dependecies
-        if: steps.poetry-cache.outputs.cache-hit != 'true'
-        run: poetry install --no-interaction --no-root
-
-      - name: Run formatter
-        run: poetry run black --check .
+      - id: deploy
```
Comment (angelocatalani) [inline .github/workflows/main.yaml:17]:
```suggestion
      - id: build application
```

File: .github/workflows/main.yaml [hunk 5]
```diff
@@ -7,47 +7,30 @@ on:
...
          poetry run pytest
+
+  format:
+    runs-on:  macos-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - id: deploy
```
Comment (angelocatalani) [inline .github/workflows/main.yaml:31]:
```suggestion
      - id: build application
```

File: .github/workflows/main.yaml [hunk 6]
```diff
@@ -7,47 +7,30 @@ on:
...
+        uses: ./.github/actions/poetry
+
+      - name: Run formatter
+        run: poetry run black --check .
+
```
Comment (angelocatalani) [inline .github/workflows/main.yaml:36]:
```suggestion
```

**Review by angelocatalani — CHANGES_REQUESTED** [review]

File: .github/actions/poetry/action.yaml (hunk 3 above)
Comment (angelocatalani) [inline .github/actions/poetry/action.yaml:36]:
```suggestion
          shell: bash

```

**Review by angelocatalani — CHANGES_REQUESTED** [review]

**Comment by soapwaster** [comment]
lgtm

**Review by soapwaster — COMMENTED** [review]
wdyt

**Review by soapwaster — COMMENTED** [review]
wdyt

**Review by angelocatalani — APPROVED** [review]

**Pull request merged**